File: server/src/main/java/pl/ordovita/surveys/domain/model/userResponse/UserResponse.java
package pl.ordovita.surveys.domain.model.userResponse;

import pl.ordovita.identity.domain.model.user.UserId;
import pl.ordovita.surveys.domain.exception.UserResponseException;
import pl.ordovita.surveys.domain.model.questions.QuestionId;

import java.time.Instant;
import java.util.List;

public class UserResponse {

    private final UserResponseId id;
    private final UserId userId;
    private final QuestionId questionId;
    private final TextAnswer textAnswer;
    private final Instant createdAt;
    private final Instant updatedAt;

    public UserResponse(UserResponseId id, UserId userId, QuestionId questionId, TextAnswer textAnswer, Instant createdAt, Instant updatedAt) {
        if (id==null) throw new UserResponseException("id cannot be null");
        if (userId==null) throw new UserResponseException("userId cannot be null");
        if (questionId==null) throw new UserResponseException("questionId cannot be null");
        if (textAnswer==null) throw new UserResponseException("textAnswer cannot be null");
        if (createdAt==null) throw new UserResponseException("createdAt cannot be null");
        if (updatedAt==null) throw new UserResponseException("updatedAt cannot be null");
        this.id = id;
        this.userId = userId;
        this.questionId = questionId;
        this.textAnswer = textAnswer;
        this.createdAt = createdAt;
        this.updatedAt = updatedAt;
    }

    public static UserResponse create(UserId userId, QuestionId questionId, TextAnswer textAnswer) {
        return new UserResponse(UserResponseId.generate(), userId, questionId, textAnswer, Instant.now(), Instant.now());
    }

    //TODO
//    public boolean checkIfAlreadyAnswered(List<QuestionId> questionIdList) {
//        if (questionIdList.contains(this.questionId)) return false;
//    }


    public UserResponseId getId() {
        return id;
    }

    public UserId getUserId() {
        return userId;
    }

    public QuestionId getQuestionId() {
        return questionId;
    }

    public TextAnswer getTextAnswer() {
        return textAnswer;
    }

    public Instant getCreatedAt() {
        return createdAt;
    }

    public Instant getUpdatedAt() {
        return updatedAt;
    }
}

File: DotNetServer/Ordovita.Domain/Surveys/UserResponse/UserResponse.cs
using Ordovita.Domain.Common;
using Ordovita.Domain.Identity;
using Ordovita.Domain.Surveys.Exception;
using Ordovita.Domain.Surveys.Questions;

namespace Ordovita.Domain.Surveys.UserResponse;

public class UserResponse : AggregateRoot<UserResponseId>
{
    public UserId UserId { get; private set; }
    public QuestionId QuestionId { get; private set; }
    public TextAnswer TextAnswer { get; private set; }
    public DateTime CreatedAt { get; private set; }
    public DateTime UpdatedAt { get; private set; }

    private UserResponse()
    {
    }

    public static Result<UserResponse> Create(UserId userId, QuestionId questionId, TextAnswer textAnswer)
    {
        if (userId.Value == Guid.Empty) return Result.Failure<UserResponse>(UserResponseException.MissingUserId);
        if (questionId.Value == Guid.Empty)
            return Result.Failure<UserResponse>(UserResponseException.MissingQuestionId);
        if (string.IsNullOrWhiteSpace(textAnswer.ToString()))
            return Result.Failure<UserResponse>(UserResponseException.MissingTextAnswer);

        var userResponse = new UserResponse
        {
            Id = UserResponseId.New(),
            UserId = userId,
            QuestionId = questionId,
            TextAnswer = textAnswer,
            CreatedAt = DateTime.UtcNow,
            UpdatedAt = DateTime.UtcNow
        };

        return Result.Success(userResponse);
    }

    public Result ChangeResponse(TextAnswer textAnswer)
    {
        if (TextAnswer == textAnswer) return Result.Failure(UserResponseException.SameTextAnswer);
        if (string.IsNullOrWhiteSpace(textAnswer.ToString()))
            return Result.Failure(UserResponseException.MissingTextAnswer);

        TextAnswer = textAnswer;
        UpdatedAt = DateTime.UtcNow;
        return Result.Success();
    }
}